### DISCUSSION TITLE
Scene Viewer vs WebXR - estimated lighting

### DISCUSSION THREAD

**Discussion by maciekglowka** [google/model-viewer#1800] [Q&A] [answered]
Hi!

I am having a hard time at the moment choosing between primary AR mode for Android devices. It seems that WebXR implementation is not supporting the light estimation based on the surroundings - at least on the devices that I have tested.
On the other hand WebXR works more smoothly than Scene Viewer (especially on lower-end devices since last AR Core update in December)
So the real question is - are there any plans to include light estimation in WebXR anytime soon (I believe specification allows it)?
Or maybe poor SceneViewer performance is a temporary glitch that might be fixed soon with an update? (although it is not a responsibility of the model-viewer team obviously)

Thanks,
Maciek

**Comment by elalish** [comment, marked as answer]
Indeed, WebXR is working on lighting estimation and we intend to ship it in model-viewer. However, we've actually gotten a lot of complaints (#871, #964, #1171) about sceneViewer's estimated lighting (which will be basically the same as WebXR's since they both come from ARCore) being often too dark or making shiny objects look matte. Keep in mind that the estimation is *very* approximate, and not always ideal (QuickLook also struggles with this).

**Reply by Lemovision** [reply]
Hi any news about lighting estimation for webxr mode in model viewer?
I think the API might be out by now if I understand correctly?
https://www.youtube.com/watch?v=JB2OyMdC4Q0
https://www.w3.org/TR/2021/WD-webxr-lighting-estimation-1-20210909/

Btw if you could find a way to make a smooth fade transition between each env map update that would be great
It's the same problem with Scene Viewer and as seen in the video above, the reflection map update looks jarring
Not sure but last time I tried on a friend's iphone, I think quick view transitions smoothly

Anyway I know it's not perfect but would still be nice to have the option in webxr
Both lighting estimation, and snapshot recording features in webxr if possible
I realized SceneView is not good for my use because I can't swap textures and have to re-download model

**Reply by elalish** [reply]
yes, the `xr-environment` attribute is now working to enable this in WebXR mode. I was about  to  link to the docs, but realized  I forgot to document it, which is probably why you're asking 🙃 I'll go add  that now.

Agreed that ARKit does a much nicer job with environment estimation than ARCore does. There's not a lot I can do about it, unfortunately. The bigger problem is it's often way too dark. I'll keep  pestering the ARCore  team.

**Reply by Lemovision** [reply]
Cool thanks for the update I'll give it a try
Any news as well about the ability to take photo/video in webxr mode?
I saw in some other old posts you were working on this as well

**Reply by elalish** [reply]
Still  waiting  on a web API for  that, but it's in progress.

**Comment by maciekglowka** [comment]
Thanks!
In my case actually it is noticeably better with the light estimation than without. (but the looks are different depending on the device)
So I think I would stick to Scene Viewer for now, and hope the glitches/delays get fixed.
If the estimated lighting is problematic for some though, maybe it could be implemented as an option / flag?
